py/picca/io.py

```diff
@@ -349,14 +361,28 @@ def read_data(in_dir,
 
     # read data taking the mode into account
     blinding = "none"
-    if mode in ["desi_mocks","desi","desi_survey_tilebased", "spcframe", "spplate", "spec", "corrected-spec"]:
-        if mode in ["desi", 'desi_mocks']:
-            pix_data, is_mock = read_from_desi(in_dir, catalog, pk1d=pk1d)
-            if (not is_mock) and ('TILEID' in catalog.colnames) and np.any((catalog['TILEID']<60000)&(catalog['TILEID']>=1000)):
-                print("you are trying to run on DESI survey tiles!")
+    if mode in ["desi_mock","desi_healpix","desi","desi_survey_tilebased", "spcframe", "spplate", "spec", "corrected-spec"]:
+        if mode in ["desi_mock", "desi"]: #I still don't think we need two different modes since we are checking if truth files exist...
+            desi_nside = 16
+            desi_prefix = f'spectra-{desi_nside}'
+            pix_data, is_mock = read_from_desi(in_dir, catalog, desi_prefix, desi_nside, pk1d=pk1d)
```

Comment:
Should we also check here for blinding? This is supposed to be a mode to read DESI data, but we are not checking for blinding. If we are sure that this is only ever going to be used for mocks, then I'd suggest deleting the mode "desi" and only keeping "desi-mock", "desi-healpix" and "desi-survey-tilebased"

Reply:
Thanks for catching that the blinding check was not indented correctly, the intent is to always check for blinding. I'll fix it. I'm also in favor to deprecate "desi" mode, but I'm not sure if p1d uses it, so I'd leave it for now an open an issue...

Reply:
OK, I ended up checking for blinding since just fixing the indentation was not the best way to do it.